modules/redis/scanner.go

```diff
@@ -228,43 +310,64 @@ func (scanner *Scanner) Scan(target zgrab2.ScanTarget) (zgrab2.ScanStatus, inter
 	}
 	defer scan.Close()
 	result := scan.result
-	pingResponse, err := scan.SendCommand("PING")
+	pingResponse, err := scan.SendCommand(scanner.commandMappings["PING"].(string))
 	if err != nil {
-		// if the first command fails (as opposed to succeeding but returning an
+		// If the first command fails (as opposed to succeeding but returning an
 		// ErrorMessage response), then flag the probe as having failed.
 		return zgrab2.TryGetScanStatus(err), nil, err
 	}
 	// From this point forward, we always return a non-nil result, implying that
 	// we have positively identified that a redis service is present.
 	result.PingResponse = forceToString(pingResponse)
 	if scanner.config.Password != "" {
-		authResponse, err := scan.SendCommand(scanner.config.AuthCommand, scanner.config.Password)
+		authResponse, err := scan.SendCommand(scanner.commandMappings["AUTH"].(string), scanner.config.Password)
 		if err != nil {
 			return zgrab2.TryGetScanStatus(err), result, err
 		}
 		result.AuthResponse = forceToString(authResponse)
 	}
-	infoResponse, err := scan.SendCommand("INFO")
+	infoResponse, err := scan.SendCommand(scanner.commandMappings["INFO"].(string))
 	if err != nil {
 		return zgrab2.TryGetScanStatus(err), result, err
 	}
 	result.InfoResponse = forceToString(infoResponse)
-	infoResponseBulk, ok := infoResponse.(BulkString)
-	if ok {
+	if infoResponseBulk, ok := infoResponse.(BulkString); ok {
+		version_found, os_found := false, false
 		for _, line := range strings.Split(string(infoResponseBulk), "\r\n") {
 			if strings.HasPrefix(line, "redis_version:") {
 				result.Version = strings.SplitN(line, ":", 2)[1]
+				version_found = true
+			} else if strings.HasPrefix(line, "os:") {
+				result.OS = strings.SplitN(line, ":", 2)[1]
+				os_found = true
+			}
+			if version_found && os_found {
 				break
 			}
 		}
 	}
-	bogusResponse, err := scan.SendCommand("NONEXISTENT")
+	bogusResponse, err := scan.SendCommand(scanner.commandMappings["NONEXISTENT"].(string))
 	if err != nil {
 		return zgrab2.TryGetScanStatus(err), result, err
 	}
 	result.NonexistentResponse = forceToString(bogusResponse)
-	quitResponse, err := scan.SendCommand("QUIT")
-	if err != nil && err != io.EOF {
+	for i := range scanner.customCommands {
+		full_cmd := strings.Fields(scanner.customCommands[i])
+		resp, err := scan.SendCommand(full_cmd[0], full_cmd[1:]...)
+		if err != nil {
+			return zgrab2.TryGetScanStatus(err), result, err
+		}
+		customResponse := CustomResponse{
+			Command:   full_cmd[0],
+			Arguments: strings.Join(full_cmd[1:], " "),
+			Response:  forceToString(resp),
+		}
+		result.CustomResponses = append(result.CustomResponses, customResponse)
+	}
+	quitResponse, err := scan.SendCommand(scanner.commandMappings["QUIT"].(string))
+	if err == io.EOF && quitResponse == nil {
+		quitResponse = NullValue
+	} else if err != nil {
```

Comment:
So this misses the case where it returns EOF *and* some data, which doesn't seem like it should be an error; i.e. `err == io.EOF` in response to a `QUIT` should be the same as `err == nil`.

Should this instead be
```suggestion
	} else if err != nil && err != io.EOF {
```

Reply:
I changed the logic so that we don't have to check whether `err == io.ERR`. Could you check the updates and let me know if that still works as intended? If not, I can use your suggestion